xbmc/interfaces/json-rpc/AddonsOperations.cpp

```diff
@@ -79,19 +80,29 @@ JSONRPC_STATUS CAddonsOperations::GetAddons(const std::string &method, ITranspor
     if (*typeIt == ADDON_UNKNOWN)
     {
       if (!enabled.isBoolean()) //All
-        CAddonMgr::GetInstance().GetInstalledAddons(typeAddons);
-      else if (enabled.asBoolean()) //Enabled
+      {
+        if (!installed.isBoolean() || installed.asBoolean())
+          CAddonMgr::GetInstance().GetInstalledAddons(typeAddons);
+        if (!installed.isBoolean() || (installed.isBoolean() && !installed.asBoolean()))
+          CAddonMgr::GetInstance().GetInstallableAddons(typeAddons);
+      }
+      else if (enabled.asBoolean() && (!installed.isBoolean() || installed.asBoolean())) //Enabled
         CAddonMgr::GetInstance().GetAddons(typeAddons);
-      else
+      else if (!installed.isBoolean() || installed.asBoolean())
```

Comment:
Yes this makes no sense to use that combination, but I choose to be respectful of the non sense and as such respect the params. This will obviously return no addons.